### PR TITLE
Chase symlinks testing

### PR DESCRIPTION
**Pull request by keszybz**
Some fallout from #13119. In the end I those changes are not needed there, so I split this out to a separate item.

### CODE REVIEW

File: src/test/test-chase-symlinks.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "", options, NULL)) >= 0)
+                switch (c) {
+
+                case 'h':
+                        log_info("Syntax:\n"
+                                 "  %s [OPTION...] path...", argv[0]);
+                        log_info("Options:");
+                        for (unsigned i = 0; i < ELEMENTSOF(options) - 1; i++)
```
Comment (poettering) [inline src/test/test-chase-symlinks.c]: doesn't matter, but ELEMENTSOF returns size_t, hence i should probably be that too

File: src/test/test-chase-symlinks.c [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "", options, NULL)) >= 0)
+                switch (c) {
+
+                case 'h':
+                        log_info("Syntax:\n"
+                                 "  %s [OPTION...] path...", argv[0]);
+                        log_info("Options:");
```
Comment (poettering) [inline src/test/test-chase-symlinks.c]: unless I am confused we usually print help texts to stdout, and this would print them to stderr

**Comment by poettering** [comment]
lgtm

File: src/test/test-chase-symlinks.c (hunk 1 above)
Comment (keszybz) [reply src/test/test-chase-symlinks.c]: Well, it's a static table a few lines up, I think unsigned is good enough ;)

File: src/test/test-chase-symlinks.c (hunk 2 above)
Comment (keszybz) [reply src/test/test-chase-symlinks.c]: Indeed, will fix.

**Comment by keszybz** [comment]
Updated with the two suggested changes. Taking liberty to set the flag.

**Pull request merged**